### ISSUE TITLE
Integration tests for Extra properties in CRs are not working

### ISSUE THREAD

**Issue by Frawless** [strimzi/strimzi-kafka-operator#4606]
With the upgrade of CRDs to v1 we started to see unexpected failures in ITs. In general, for tests where we are trying to deploy CRs with some extra parameters which are not part of the CRD scheme, we expect that kubectl call to API server will fail and the CR won't be created. However, the CR is created and extra properties are just stripped away by api server (probably).

Audit logs of api server says that CR was created/updated and extra args are not there. For example:

```
{
   "kind":"Event",
   "apiVersion":"audit.k8s.io/v1",
   "level":"RequestResponse",
   "auditID":"2807991c-1391-499e-9a0a-25726e1a7131",
   "stage":"ResponseComplete",
   "requestURI":"/apis/kafka.strimzi.io/v1beta2/namespaces/default/kafkaconnects/test-kafka-connect?fieldManager=kubectl-client-side-apply",
   "verb":"patch",
   "user":{
      "username":"admin",
      "uid":"b34d8ff3-5b35-48ba-8887-6730658134f9",
      "groups":[
         "system:authenticated:oauth",
         "system:authenticated"
      ],
      "extra":{
         "scopes.authorization.openshift.io":[
            "user:full"
         ]
      }
   },
   "sourceIPs":[
      "192.168.1.126"
   ],
   "userAgent":"oc/4.7.0 (darwin/amd64) kubernetes/c66c03f",
   "objectRef":{
      "resource":"kafkaconnects",
      "namespace":"default",
      "name":"test-kafka-connect",
      "apiGroup":"kafka.strimzi.io",
      "apiVersion":"v1beta2"
   },
   "responseStatus":{
      "metadata":{
         
      },
      "code":200
   },
   "requestObject":{
      "extra":true,
      "jkk":"jhg",
      "spec":{
         "extra":true,
         "kimage":"foo",
         "kreplicas":6
      }
   },
   "responseObject":{
      "apiVersion":"kafka.strimzi.io/v1beta2",
      "kind":"KafkaConnect",
      "metadata":{
         "annotations":{
            "kubectl.kubernetes.io/last-applied-configuration":"{\"apiVersion\":\"kafka.strimzi.io/v1beta2\",\"extra\":true,\"jkk\":\"jhg\",\"kind\":\"KafkaConnect\",\"metadata\":{\"annotations\":{},\"name\":\"test-kafka-connect\",\"namespace\":\"default\"},\"spec\":{\"bootstrapServers\":\"kafka:9092\",\"config\":{\"name\":\"bar\"},\"extra\":true,\"kimage\":\"foo\",\"kreplicas\":6}}\n"
         },
         "creationTimestamp":"2021-03-19T14:53:09Z",
         "generation":1,
         "name":"test-kafka-connect",
         "namespace":"default",
         "resourceVersion":"4916884",
         "selfLink":"/apis/kafka.strimzi.io/v1beta2/namespaces/default/kafkaconnects/test-kafka-connect",
         "uid":"3836b52c-35e3-4863-b896-2a7c0c39dff6"
      },
      "spec":{
         "bootstrapServers":"kafka:9092",
         "config":{
            "name":"bar"
         }
      }
   },
   "requestReceivedTimestamp":"2021-03-19T14:59:13.339180Z",
   "stageTimestamp":"2021-03-19T14:59:13.358682Z",
   "annotations":{
      "authorization.k8s.io/decision":"allow",
      "authorization.k8s.io/reason":"RBAC: allowed by ClusterRoleBinding \"cluster-admin\" of ClusterRole \"cluster-admin\" to User \"admin\""
   }
}
```

From my POV it's bug/feature in kube. Until we will find more, I am going to disable the tests. We should track progress here.

**Comment by scholzj** [comment]
@Frawless Should this be solved by #4595? Or is there something more to do here?

**Comment by scholzj** [comment]
@Frawless Never mind -> it just disabled the tests, so this is still open ;-)
